### PR TITLE
feat: Check for internet connectivity when upgrading/installing

### PR DESCRIPTION
**Pull request by tarungarg546**
:baby:

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -3,31 +3,48 @@
...
 			spawn(`${__dirname}/node-nightly/node`, args, {stdio: 'inherit', env: process.env});
 		} else{
-			spawn(`${__dirname}/node-nightly/bin/node`, args, {stdio: 'inherit', env: process.env});			
+			spawn(`${__dirname}/node-nightly/bin/node`, args, {stdio: 'inherit', env: process.env});
 		}
 	}).catch(console.error);
 }
+
+function reportIfOffline() {
+	isOnline((err, online) => {
```
Comment (hemanth) [inline cli.js]: `err` not used ;)

**Comment by hemanth** [comment]
Nice.

File: cli.js (hunk 1 above)
Comment (tarungarg546) [inline cli.js]: According to [documentation](https://www.npmjs.com/package/is-online), `err` is just for keeping convention going :palm_tree: .

Comment (hemanth) [inline cli.js]: let us promisify it :D

Comment (tarungarg546) [inline cli.js]: Keep it simple :lollipop:

**Pull request merged**